source/_ecosystem/oauth2.markdown

```diff
@@ -0,0 +1,202 @@
+
+1. Create a new project: https://console.developers.google.com/project
+2. Choose the new project from the top right project dropdown (only if another project is selected)
+3. In the project Dashboard center pane, choose **"Enable and manage APIs"**
+4. In the left Nav pane, choose **"Credentials"**
+5. In the center pane, choose **"OAuth consent screen"** tab. Fill in **"Product name shown to users"** and hit save.
+6. In the center pane, choose **"Credentials"** tab.
+   * Open the **"New credentials"** drop down
+   * Choose **"OAuth client ID"**
+   * Choose **"Web application"**
+   * Application name is freeform, choose something appropriate
+   * Authorized JavaScript origins is your domain ex: `https://internal.yourcompany.com`
+   * Authorized redirect URIs is the location of oath2/callback ex: `https://internal.yourcompany.com/oauth2/callback`
+   * Choose **"Create"**
+4. Take note of the **Client ID** and **Client Secret**
+
+### {% linkable_title 2. Install oauth2_proxy %}
+
+I am going to take the cheap route and send you directly to the [Github Page](https://github.com/bitly/oauth2_proxy) for oauth2_proxy for general installation of the product.
+
+If you are using docker, you can use this command to run it and reference the configuration files we create in the next step:
+
+```
+ docker run -d --name="oauth2_proxy" --net="bridge" -e -p 8095:8095/tcp -v "/path/to/appdata/oauth2_proxy":"/config":rw bitly/oauth2_proxy -config="/config/oauth2_proxy.cfg"
+```
+
+### {% linkable_title 3. Configure oauth2_proxy %}
+
+We are going to create 2 files, "oauth2_proxy.cfg" and "auth_emails", which will contain a list of email addresses that will be allowed in.
+
+In oauth2_proxy.cfg, the biggest non-standard thing we are going to do is to proxy it to file:///dev/null so that it doesn't attempt to return anything other than the 200 OK response to nginx because oauth2_proxy is only being used to auth, not to proxy.
+
+```
+http_address = "0.0.0.0:8095"
+upstreams = [
+    "file:///dev/null"
+ ]
+client_id = "Client_ID"
+client_secret = "Client_Secret"
+authenticated_emails_file = "/config/auth_emails"
+cookie_name = "_home_assistant"
+cookie_secret = "4C4F1B247E837B62"
+cookie_domain = "example.com"
+cookie_expire = "72h"
+cookie_refresh = "4h"
+cookie_secure = false
+```
+
+### {% linkable_title 4. Edit your nginx configuration %}
+
+We need to add several things to the nginx configuration. If you are currently using basic authentication with nginx, you probably want to disable them at this point or you are going to have several separate layers to sign in.
+
+### {% linkable_title Oauth2 Locations %}
+
+We need to create locations to proxy the /oauth2 commands through nginx when they are requested.
+```
+  location = /oauth2/auth {
+      internal;
+      proxy_pass http://10.10.10.200:8095;
+      proxy_set_header Host $host;
+      proxy_pass_request_body off;
+      proxy_set_header Content-Length "";
```

Comment:
I was actually wrong. The /oauth2/auth request is special, and wouldn't return anything anyway. the `/dev/null` proxy is just housekeeping because it requires _something_ to be put in the upstreams.
Write value, wrong reason.